external-crates/move/crates/move-package-alt/src/package/package_impl.rs

```diff
@@ -351,8 +351,19 @@ fn check_for_environment<F: MoveFlavor>(
     manifest: &Manifest,
     env: &EnvironmentName,
 ) -> PackageResult<()> {
-    let mut known_environments = manifest.environments();
-    known_environments.extend(F::default_environments());
+    let mut known_environments = F::default_environments();
+    let manifest_envs = manifest.environments();
+
+    if let Some((name, _)) = manifest_envs.iter().find(|(name, chain_id)| {
+        known_environments
+            .get(*name)
+            .is_some_and(|default| default != *chain_id)
```

Comment:
I wonder if we just want to disallow it entirely instead of allowing people to write in the actual chain IDs?